hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotFileCache.java

```diff
@@ -226,7 +226,7 @@ public synchronized void triggerCacheRefreshForTesting() {
     return unReferencedFiles;
   }
 
-  private void refreshCache() throws IOException {
+  private synchronized void refreshCache() throws IOException {
```

Comment:
We do not need to make it synchronized? We will only call it below in RefreshCacheTask and we have synchronized there.

Reply:
This method is also called in getUnreferencedFiles, which is used in concurrently checking deletable files. I kept the synchronize in RefreshCacheTask, because it will clear the cache when encountered exception. But when in getUnreferencedFiles the exception should throw out.

Reply:
It's not necessary to make this method synchronized, I have changed it.